prov/util/src/util_mem_hooks.c

```diff
@@ -433,6 +438,28 @@ static int ofi_intercept_shmdt(const void *shmaddr)
 	return real_calls.shmdt(shmaddr);
 }
 
+static int ofi_intercept_brk(const void *brkaddr)
+{
+	void *old_addr, *new_addr;
+
+	FI_DBG(&core_prov, FI_LOG_MR,
+	      "intercepted brk addr %p\n", brkaddr);
+
+	old_addr = sbrk (0);
+	new_addr = sbrk (0);
```

Comment:
Doesn't this set old and new address to the same value?

Reply:
@shefty actually, yes. i left it temporarily. We need to find the `new_addr` like that: [here](https://github.com/open-mpi/ompi/blob/68ef097f1d81da5144d5ade43143ae9eed5a1f27/opal/mca/memory/patcher/memory_patcher_component.c#L304)

Reply:
@shefty Could we set `new_addr = brkaddr`?

Reply:
I would copy the OMPI implementation.  I skipped it when adding the other hooks because I didn't think it was worth it.  But it sounds like it is needed still.

Reply:
@shefty Tried to do the same. The OMPI implementation without _CURBRK, I guess.